product-service/handlers/getProductById.ts

```diff
@@ -1,22 +1,15 @@
-import { Context, Callback } from 'aws-lambda';
-import { ProductService } from '../services';
+import { createResponse, ProductService, STATUS_CODE } from '../services';
 
 const productService = new ProductService();
-export async function getProductById(event: any, context: Context, callback: Callback) {
+export async function getProductById(event: any) {
 
+  try {
     const { productid } = event.pathParameters;
-  const data = await productService.getProductById(productid);
-  const response = {
-    statusCode: 200,
-    headers: {
-        'Access-Control-Allow-Methods': '*',
-        'Access-Control-Allow-Headers': '*',
-        'Access-Control-Allow-Origin': '*'
-    },
-    body: JSON.stringify({
-        ...data,
-    }),
-  };
-  
-  return response;
+    const data: any = await productService.getProductById(productid);
+    const response = createResponse(data, false, STATUS_CODE.SUCCESS);
+    return response
+  } catch(err: any) {
+    const response = createResponse(err, true, err.errorCode ?? STATUS_CODE.INTERNAL_ERROR);
```

Comment:
Please pass err.message also as fourth argument of createResponse otherwise it'll return unexpected error occurred message which is not correct for product not found scenario.

Reply:
<img width="711" alt="image" src="https://user-images.githubusercontent.com/10434846/196105650-505b9887-a1d4-4313-b72d-f076d39fb29b.png">

As of now I am using error object to send all this information. I will add same message to body as well

Reply:
Added error message to body as well